pkg/algorithm/utils.go

```diff
@@ -169,26 +169,29 @@ func generateAffinityGroupBindInfo(
 	return affinityGroupBindInfo, selectedNode, selectedGpuIndices, chain
 }
 
-// collectNodesNotSuggested collects all the nodes that are not within the suggested nodes
+// collectBadOrNonSuggestedNodes collects all the nodes that are not within the suggested nodes
 // in the physical placement of an affinity group.
-func collectNodesNotSuggested(
-	placement groupPhysicalPlacement, suggestedNodes common.Set) (nodesNotInSuggested common.Set) {
+func collectBadOrNonSuggestedNodes(
+	placement groupPhysicalPlacement,
+	suggestedNodes common.Set) (
+	badOrNonSuggestedNodes common.Set) {
 
-	nodesNotInSuggested = common.NewSet()
+	badOrNonSuggestedNodes = common.NewSet()
 	for gpuNum := range placement {
 		for podIndex := range placement[gpuNum] {
 			for _, gpu := range placement[gpuNum][podIndex] {
 				if gpu == nil {
 					continue
 				}
 				nodes, _ := gpu.(*PhysicalCell).GetPhysicalPlacement()
-				if !suggestedNodes.Contains(nodes[0]) {
-					nodesNotInSuggested.Add(nodes[0])
+				if gpu.(*PhysicalCell).GetAPIStatus().CellHealthiness == api.CellBad ||
```

Comment:
>GetAPIStatus [](start = 27, length = 12)

The APIStatus now not only for expose , but also impacts our own scheduling logic.
So, maybe we need to rename APIStatus to other name. And adjust previous comments, that it is **not** only for expose

Reply:
I created another field in Cell structure (Cell.healthy) so we no longer need to access APIStatus to know if a cell is healthy. So I think we can keep this name